### PR TITLE
[maintenance] remove c_str usage from StringUtils::Format calls

### PR DESCRIPTION
**Pull request by lrusak**
This is a follow up to #19684 

I've done a few things. The first commit is an automated cleanup of `.c_str()` usage from `StringUtils::Format` calls. This is done by the following:
```
sed -i "${FILE}" \
-e ':a
    /StringUtils::Format(/{
      :line
      s/.c_str()//g
      /);/!{
        N
        bline
      }
      N
    }'
```

The weird sed syntax is so I don't have to use `;` line breaks and so that I can make sure I match multiline calls.

The following commits prefixed with `[SQUASH]` are commits I made manually to fix the build. I may need more of these depending on what jenkins says.

I've also included a commit to remove redundant usage of `StringUtils::Format` and simplified it to use std::to_string() where applicable. This was an automated commit using
```
sed -i "${FILE}" -E \
-e '/StringUtils::Format\("\{\}",/{
      /fmt::ptr/b
      s/StringUtils::Format\("\{\}", (.*)\)/std::to_string\(\1\)/g
    }'
```

I've also included a manual commit after his that fixes the build after the `std::to_string` changes as some methods return a `std::string` already!

It's amazing to me after years of refactoring that some of the boilerplate can be removed completely.

### CODE REVIEW

**Review by garbear — APPROVED** [review]

File: xbmc/utils/log.h [hunk 1]
```diff
@@ -155,9 +155,9 @@ class CLog : public ISettingsHandler, public ISettingCallback
                                                   const char* format,
                                                   Args&&... args)
   {
-    GetInstance().FormatAndLogInternal(level,
-                                       StringUtils::Format("{0:s}: {1:s}", functionName, format),
-                                       std::forward<Args>(args)...);
+    GetInstance().FormatAndLogInternal(
+        level, StringUtils::Format("{0:s}: {1:s}", functionName, format).c_str(),
```
Comment (Montellese) [inline xbmc/utils/log.h]: Looking at this I'm wondering why `FormatAndLogInternal()` takes a `const char*` / `const wchar*` instead of a `const std::string&` / `const std::wstring&` because `StringUtils::Format()` does. I'll take a look at that ASAP.

File: xbmc/cdrip/CDDARipJob.cpp [hunk 2]
```diff
@@ -192,8 +192,8 @@ CEncoder* CCDDARipJob::SetupEncoder(CFile& reader)
     return NULL;
 
   // we have to set the tags before we init the Encoder
-  const std::string strTrack =
-      StringUtils::Format("{}", strtol(m_input.substr(13, m_input.size() - 13 - 5), nullptr, 10));
+  const std::string strTrack = StringUtils::Format(
```
Comment (Montellese) [inline xbmc/cdrip/CDDARipJob.cpp:195]: This looks odd. Why do we first convert a string to an integer and then format it back to a string? Is this on purpose to remove any non-digit characters? Otherwise `const auto strTrack = m_input.substr(13, m_input.size() - 13 - 5);` would be enough.

File: xbmc/utils/SortUtils.cpp [hunk 3]
```diff
@@ -247,11 +247,11 @@ std::string ByYear(SortAttribute attributes, const SortItem &values)
 }
 
 std::string ByOrigDate(SortAttribute attributes, const SortItem& values)
-{
```
Comment (Montellese) [inline xbmc/utils/SortUtils.cpp:250]: Unrelated

File: xbmc/utils/SortUtils.cpp [hunk 4]
```diff
@@ -247,11 +247,11 @@ std::string ByYear(SortAttribute attributes, const SortItem &values)
 }
 
 std::string ByOrigDate(SortAttribute attributes, const SortItem& values)
-{  
+{
   std::string label;
   label = values.at(FieldOrigDate).asString();
 
-  const CVariant &album = values.at(FieldAlbum);
```
Comment (Montellese) [inline xbmc/utils/SortUtils.cpp:254]: Unrelated

**Review by Montellese — APPROVED** [review]
Looks good in general. Thanks for taking the time to clean this up.

File: xbmc/utils/log.h (hunk 1 above)
Comment (Montellese) [reply xbmc/utils/log.h]: OK great, so I don't have to look into it :+1:

Comment (lrusak) [reply xbmc/utils/log.h]: I'll look for your opinion when I PR the changes 😄

File: xbmc/cdrip/CDDARipJob.cpp (hunk 2 above)
Comment (lrusak) [reply xbmc/cdrip/CDDARipJob.cpp:195]: I'm really not sure 😕

File: xbmc/peripherals/devices/PeripheralCecAdapter.cpp [hunk 5]
```diff
@@ -1588,7 +1587,7 @@ std::string CPeripheralCecAdapterUpdateThread::UpdateAudioSystemStatus(void)
     CLog::Log(LOGDEBUG,
               "%s - CEC capable amplifier found (%s). volume will be controlled on the amp",
               __FUNCTION__, ampName.c_str());
-    strAmpName += StringUtils::Format("{}", ampName.c_str());
+    strAmpName += std::to_string(ampName);
```
[the remark this answers is not in the document]
Comment (fuzzard) [reply xbmc/peripherals/devices/PeripheralCecAdapter.cpp:1590]: Haha, looked again at this, it doesnt need the std::to_string call, ampName is already a string, just do 
```
strAmpName += ampName;
```

or whatever you are comfortable with

File: xbmc/platform/darwin/osx/peripherals/PeripheralBusUSB.cpp [hunk 6]
```diff
@@ -237,10 +237,9 @@ void CPeripheralBusUSB::DeviceAttachCallback(CPeripheralBusUSB* refCon, io_itera
           }
         }
         if (!ttlDeviceFilePath.empty())
-          privateDataRef->result.m_strLocation =
-              StringUtils::Format("{}", ttlDeviceFilePath.c_str());
+          privateDataRef->result.m_strLocation = std::to_string(ttlDeviceFilePath);
```
Comment (fuzzard) [inline xbmc/platform/darwin/osx/peripherals/PeripheralBusUSB.cpp:240]: Same as the other one, ttlDeviceFilePath is already a string, drop the to_string()

**Comment by fuzzard** [comment]
From an OSX perspective, with those 2 changes mentioned, it builds ok

File: xbmc/cdrip/CDDARipJob.cpp (hunk 2 above)
Comment (DaveTBlake) [reply xbmc/cdrip/CDDARipJob.cpp:195]: CD ripping in Kodi is ancient. My guess would be to remove things like leading zeros or non-digits, but I don't know either despite it being music related. 

I actually think Kodi should not be a ripper at all, there is so many better software tools for that out there. We want users to fully tag their music, even best (for efficient scraping of extra art etc.) with Musicbranz ids, and Kodi can't do that just offers basic CDDA (track naming based). Sorry a digression!

**Review by DaveTBlake — APPROVED** [review]
Visual inspection of music related code all looks fine to me.

**Comment by lrusak** [comment]
@Montellese I pushed a fixup as we have some tests that rely on the StringUtils::Format class enum conversion. Is this behaviour actually used anywhere?

**Comment by Montellese** [comment]
> @Montellese I pushed a fixup as we have some tests that rely on the StringUtils::Format class enum conversion. Is this behaviour actually used anywhere?

#17318 introduced the fix and the issue e.g. exists in the video database where we format SQL queries using code like `StringUtils::Format("movie.c%02d", VIDEODB_ID_TITLE)` with `VIDEODB_ID_TITLE` being an enum value. Without the workaround this is formatted to `movie.c1` instead of `movie.c01` so yes this is still relevant.

**Comment by lrusak** [comment]
> > @Montellese I pushed a fixup as we have some tests that rely on the StringUtils::Format class enum conversion. Is this behaviour actually used anywhere?
> 
> #17318 introduced the fix and the issue e.g. exists in the video database where we format SQL queries using code like `StringUtils::Format("movie.c%02d", VIDEODB_ID_TITLE)` with `VIDEODB_ID_TITLE` being an enum value. Without the workaround this is formatted to `movie.c1` instead of `movie.c01` so yes this is still relevant.

I'm not really sure if I like this solution, It seems that recent versions of libfmt can also deal with enum class on it's own. My issue is that enum class is supposed to protect against implicit casting. If the issue is just about formatting the integer with padding that can easily be handled with `{:02}`.

**Comment by Montellese** [comment]
> I'm not really sure if I like this solution, It seems that recent versions of libfmt can also deal with enum class on it's own. My issue is that enum class is supposed to protect against implicit casting. If the issue is just about formatting the integer with padding that can easily be handled with `{:02}`.

Yes newer versions of `libfmt` can properly handle this but I'm not sure if such a version is available for all platforms we support (especially Ubuntu)?
With the currently used version of `libfmt` using `02` doesn't work independent of whether it's `%02d` or `{:02}` or `{:02d}`. `libfmt` doesn't categorize class enums as integers and therefore ignores any additional formating specifications like `0`, `2` or `02` etc. That's why we need the cast to `int`. But the new version of `libfmt` doesn't do anything different.

**Comment by lrusak** [comment]
> > I'm not really sure if I like this solution, It seems that recent versions of libfmt can also deal with enum class on it's own. My issue is that enum class is supposed to protect against implicit casting. If the issue is just about formatting the integer with padding that can easily be handled with `{:02}`.
> 
> Yes newer versions of `libfmt` can properly handle this but I'm not sure if such a version is available for all platforms we support (especially Ubuntu)?
> With the currently used version of `libfmt` using `02` doesn't work independent of whether it's `%02d` or `{:02}` or `{:02d}`. `libfmt` doesn't categorize class enums as integers and therefore ignores any additional formating specifications like `0`, `2` or `02` etc. That's why we need the cast to `int`. But the new version of `libfmt` doesn't do anything different.

No worries, maybe it's something to look into more in the future :smiley:

**Comment by lrusak** [comment]
g2g?

**Comment by lrusak** [comment]
I've added an extra commit that fixes a few `Format` calls that I missed in the first round
